articles/121_realtime_proposal.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: default
+title: Proposal for Implementation of Real-time Systems in ROS 2
+permalink: articles/realtime_proposal.html
+abstract: Proposal for a test-driven approach to the real-time performance requirement in ROS 2.
+published: true 
+author: Jackie Kay
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Requirements and Implementation of Real-Time Systems
+
+### System requirements
+
+The requirements of a real-time system vary depending on the use case.
+At its core, the real-time requirement of a system has two components:
+
+- Latency
+
+  - Update period
+  - Determinism (measured by jitter)
```

Comment:
Yeah, I agree, I have mostly misused the term "determinism" throughout this document. A system is either deterministic or nondeterministic. I will take a look at improving the precision of the language in this article.